### PR TITLE
travis: also test against docker 1.7, 1.9, 1.10

### PR DESCRIPTION
**Pull request by mattnworb**

### CODE REVIEW

**Comment by davidxia** [comment]
Good job on dumping the docker logs. `testIpcMode()` fails on docker 1.9+ because `/dev/mqueue` isn't mounted by default. See https://github.com/boot2docker/boot2docker/issues/1098. Should we tell travis to mount it?

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java [hunk 1]
```diff
@@ -666,7 +672,8 @@ public void testBuildName() throws Exception {
     final String dockerDirectory = Resources.getResource("dockerDirectory").getPath();
     final String imageId = sut.build(Paths.get(dockerDirectory), imageName);
     final ImageInfo info = sut.inspectImage(imageName);
-    assertThat(info.id(), startsWith(imageId));
+    final String expextedId = dockerApiVersionLessThan("1.22") ? imageId : "sha256:" + imageId;
```
Comment (davidxia) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: nit: typo in `expexted`

**Comment by davidxia** [comment]
API 1.22 doesn't throw an exception for `testNetworksConnectContainerShouldFailIfContainerNotRunning()`. Perhaps just delete this test?

**Comment by johnflavin** [comment]
Use `requireApiVersionLessThan("1.22")` (or whatever the exact name is). I added that recently to test that features still work in older versions.

**Comment by davidxia** [comment]
@johnflavin I considered that, but the test already requires the version be at least 1.21. So the test would only apply for 1.21. At that point, it seems the test isn't that valuable.

**Comment by davidxia** [comment]
`testEventStream()` hangs on docker 1.7 because docker-client is trying to read from the socket to docker daemon, but docker doesn't return bytes.

Compare 1.7

```
 ⮀ curl --insecure --cert ~/.docker/machine/machines/1.7.0-vmware/b2d-client-side.p12 --pass tcuser 'https://192.168.212.131:2376/events' -vvv
*   Trying 192.168.212.131...
* Connected to 192.168.212.131 (192.168.212.131) port 2376 (#0)
* WARNING: SSL: Certificate type not set, assuming PKCS#12 format.
* Client certificate: dxia
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
* Server certificate: dxia.1.7.0-vmware
* Server certificate: dxia
> GET /events HTTP/1.1
> Host: 192.168.212.131:2376
> User-Agent: curl/7.43.0
> Accept: */*
>
```

vs 1.10 where the test passes.

```
curl --insecure --cert ~/.docker/machine/machines/1.7.0-vmware/b2d-client-side.p12 --pass tcuser 'https://192.168.212.133:2376/events' -vvv
*   Trying 192.168.212.133...
* Connected to 192.168.212.133 (192.168.212.133) port 2376 (#0)
* WARNING: SSL: Certificate type not set, assuming PKCS#12 format.
* Client certificate: dxia
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
* Server certificate: dxia.1.10.3-vmware
* Server certificate: dxia
> GET /events HTTP/1.1
> Host: 192.168.212.133:2376
> User-Agent: curl/7.43.0
> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Type: application/json
< Server: Docker/1.10.3 (linux)
< Date: Tue, 29 Mar 2016 17:06:09 GMT
< Transfer-Encoding: chunked
<
```

**Comment by johnflavin** [comment]
I think this is a known bug in 1.7.0. This issue report ([docker/docker#14354](https://github.com/docker/docker/issues/14354)) looks a lot like what you’re seeing. There was a PR to fix it for 1.7.1 ([docker/docker#14359](https://github.com/docker/docker/pull/14359)), but if I am reading the comments correctly it was not included in that release. But, then again, maybe I'm reading them wrong. Do you know if you have 1.7.0 or 1.7.1?

**Comment by davidxia** [comment]
@johnflavin I'm using 1.7.0 locally, but travis is using 1.7.1 and has the same result: https://travis-ci.org/spotify/docker-client/jobs/119296384

**Comment by davidxia** [comment]
I just tried 1.7.1 locally and had the same issue as 1.7.0. According to https://github.com/docker/docker/pull/14359, this was removed from the 1.7.1 release. Not sure where it ended up. We could have a conditional in `testEventStream()` to take care of this.

**Comment by mattnworb** [comment]
We could also ignore this test in 1.7.x (or remove 1.7.x from Travis)

**Comment by davidxia** [comment]
@mattnworb I'll skip it for 1.7.x. The bug completely breaks `DockerClient.events()`.

**Comment by davidxia** [comment]
@mattnworb Yay, green.

**Comment by mattnworb** [comment]
wow! :tada:

**Pull request merged**